cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,472 @@
+        assumption = {
+            "SourcePrincipal": source_principal,
+            "DestinationPrincipal": destination_principal,
+            "Action": event_name,
+            "EventId": event.get("EventId"),
+            "EventTime": event.get("EventTime"),
+            "SourceIPAddress": event.get("SourceIPAddress"),
+            "UserAgent": event.get("UserAgent"),
+            "AwsRegion": event.get("AwsRegion", region),
+            "AccountId": current_aws_account_id,
+            "AssumedRoleArn": destination_principal,  # For relationship targeting
+            "PrincipalArn": source_principal,  # For relationship targeting
+            "SessionName": cloudtrail_event
+            and cloudtrail_event.get("requestParameters", {}).get("roleSessionName"),
+            "RequestId": cloudtrail_event and cloudtrail_event.get("requestID"),
+            "RecipientAccountId": cloudtrail_event
+            and cloudtrail_event.get("recipientAccountId"),
+        }
+        role_assumptions.append(assumption)
+
+    logger.info(
+        f"Successfully transformed {len(role_assumptions)} role assumptions from {len(events)} events"
+    )
+
+    # Aggregate role assumptions by (source, destination) pairs
+    aggregated_assumptions = _aggregate_role_assumptions(role_assumptions)
+
+    return aggregated_assumptions
+
+
+@timeit
+def load_role_assumptions(
+    neo4j_session: neo4j.Session,
+    aggregated_role_assumptions: List[Dict[str, Any]],
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    """
+    Load aggregated role assumption relationships into Neo4j using MatchLink pattern.
+
+    Creates direct ASSUMED_ROLE relationships with aggregated properties:
+    (AWSUser|AWSRole|AWSPrincipal)-[:ASSUMED_ROLE {lastused, times_used, first_seen, last_seen}]->(AWSRole)
+
+    Assumes that both source principals and destination roles already exist in the graph.
+
+    :type neo4j_session: neo4j.Session
+    :param neo4j_session: The Neo4j session to use for database operations
+    :type aggregated_role_assumptions: List[Dict[str, Any]]
+    :param aggregated_role_assumptions: List of aggregated role assumption relationships from transform function
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :type aws_update_tag: int
+    :param aws_update_tag: Timestamp tag for tracking data freshness
+    :rtype: None
+    """
+    # Use MatchLink to create relationships between existing nodes
+    matchlink_schema = AssumedRoleMatchLink()
+
+    load_matchlinks(
+        neo4j_session,
+        matchlink_schema,
+        aggregated_role_assumptions,
+        lastupdated=aws_update_tag,
+        _sub_resource_label="AWSAccount",
+        _sub_resource_id=current_aws_account_id,
+    )
+
+    logger.info(
+        f"Successfully loaded {len(aggregated_role_assumptions)} role assumption relationships"
+    )
+
+
+def _aggregate_role_assumptions(
```

Comment:
As mentioned above, this really should be the main logic of the `transform()` function since these are the only fields that are really needed to load things to the graph.